### ISSUE TITLE
Wrong type hint of DOMNode.watch method

### ISSUE THREAD

**Issue by mzebrak** [Textualize/textual#1805]
It looks like after the latest changes `0.11.0` (because in 0.10.1 there was no issue), the wrong type hint was used for DOMNode.watch method.

After upgrading textual in my project I see mypy errorrs like below:

```bash
Argument 3 to "watch" of "DOMNode" has incompatible type "Callable[[Any], None]"; expected "Union[Callable[[], Awaitable[None]], Callable[[], None]]"  [arg-type]
```

After some investigation, argument no.3 in the `0.11.0` is:
```python
callback: CallbackType,
```

which is:
```python
CallbackType = Union[Callable[[], Awaitable[None]], Callable[[], None]]
```

This is not true because Callabe can take additional arguments - current_value and previous_value (both of the Any type).

Previously, in the `reactive.watch` method, it was: 
```python
 callback: Callable[[Any], object]
 ```

**Comment by mzebrak** [comment]
@willmcgugan  This one is not fixed in 0.13.0 so shouldn't be closed.

**Comment by mzebrak** [comment]
0.14.0 fixed most of the errors, but there is still one minor issue:
```bash
error: Argument "callback" to "watch" of "DOMNode" has incompatible type "Callable[[], None]"; expected "Union[Callable[[Any], Awaitable[None]], Callable[[Any], None]]"  [arg-type]
```

It shows up when I attach callack method without any params:

```python
self.watch(self.app, "something", callback=self.__sync_cart_items)

def __sync_cart_items(self) -> None:
    pass
```

I have to apply workaround such as:

```python
self.watch(self.app, "something", callback=self.__sync_cart_items)

def __sync_cart_items(self, _: Something) -> None:
    pass
```
